datastore/storemanager.go

```diff
@@ -397,9 +410,63 @@ func (sm *StoreManager) queryGET(q config.MetricQuery) {
 	}
 
 	var resp config.APIQueryResponse
+	var step int
+	var table string
+	var singleStat float64
+	var statList []float64
+
+	// Initialize series map
+	series := map[string][]float64{}
+
+	// Get difference between now and q.From to determine which rollup table to query
+	timeDelta := time.Since(time.Unix(q.From, 0))
+
+	// Determine lookup table and data point step
+	for _, path := range q.Query {
+		expr := sm.getExpression(path)
+		for _, window := range config.G.Rollup[expr].Windows {
+			config.G.Log.System.LogDebug("evaluating %v", window)
+			config.G.Log.System.LogDebug("timeDelta: %d, window.Retention: %d", timeDelta.Seconds(), window.Retention.Seconds())
+			if timeDelta < window.Retention {
+				table = window.Table
+				step = int(window.Window.Seconds())
+				break
+			}
+		}
+
+		// Build query for this stat path
+		query := fmt.Sprintf(`SELECT stat FROM %s.%s WHERE path = ? AND time >= ? AND time <= ?`,
+			config.G.Cassandra.Keyspace, table)
+
+		config.G.Log.System.LogDebug("Query: %s", query)
+
+		// Populate statList with returned stats.
+		iter := sm.dbClient.Query(query, path, time.Unix(q.From, 0), time.Unix(q.To, 0)).Iter()
+		for iter.Scan(&singleStat) {
+			config.G.Log.System.LogDebug("singleStat: %d", singleStat)
+			statList = append(statList, singleStat)
+		}
+
+		config.G.Log.System.LogDebug("statList: %v", statList)
+
+		if err := iter.Close(); err != nil {
```

Comment:
This can be deferred, and moved right after the creation.
defer func() {
...
}()